### PR TITLE
Mark bundle product type doesn't support sources for all platforms

### PR DESCRIPTION
**Pull request by serejahh**
### Short description 📝

I've got a warning: `Warning: The target FMPFeedbackForm_FMPFeedbackForm doesn't contain source files.`
This is because we use https://github.com/MacPaw/FMPFeedbackForm. There's a bundle target inside:
image
I found that tuist handles iOS and tvOS bundles but doesn't handle other platforms for some reason. So I changed to `a bundle target on ANY platform cannot support sources`

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [ ] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by serejahh** [comment]
It's a macOS target. I've added a sample project to the branch. Please have a look 🙏
image

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: projects/tuist/fixtures/macos_app_with_macOS_bundle_target/Project.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import ProjectDescription
```
[the remark this answers is not in the document]
Comment (serejahh) [reply projects/tuist/fixtures/macos_app_with_macOS_bundle_target/Project.swift:1]: Sure, I'll remove the project. It's just a showcase how I get the warning

File: Sources/TuistGraph/Models/Target.swift [hunk 2]
```diff
@@ -145,8 +145,8 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
     /// Returns true if the target supports having sources.
     public var supportsSources: Bool {
         switch (platform, product) {
-        case (.iOS, .bundle), (.iOS, .stickerPackExtension),
-             (.watchOS, .watch2App), (.tvOS, .bundle):
+        case (_, .bundle), (.iOS, .stickerPackExtension),
+             (.watchOS, .watch2App):
             return false
```
[the remark this answers is not in the document]
Comment (serejahh) [reply Sources/TuistGraph/Models/Target.swift]: Updated `TargetLinter`

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 3]
```diff
@@ -96,6 +96,11 @@ class TargetLinter: TargetLinting {
         let hasNoDependencies = target.dependencies.isEmpty
         let hasNoScripts = target.scripts.isEmpty
 
+        // macOS bundle targets can have source code, but it's optional
+        if target.platform == .macOS && target.product == .bundle && hasNoSources {
+            return []
+        }
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:102]: nit: This could be a guard at the beginning of the method

**Review by danyf90 — APPROVED** [review]

**Review by kwridan — APPROVED** [review]
Thanks @serejahh 

May be worth extending `TargetLinterTests` for this case.

**Comment by serejahh** [comment]
Added a test that macOS bundle with no sources doesn't report a warning

**Pull request merged**